client/src/js/components/bhPermissionTree.js

```diff
@@ -0,0 +1,134 @@
+angular.module('bhima.components')
+  .component('bhPermissionsTree', {
+    bindings : {
+      assignedUnits : '=', // array [{unit_id :1}, {unit_id : 3}, ...]
```

Comment:
So, the best API for this is probably to have the following bindings:
```js
bindings : {
  assignedUnits : '<?',  // this could be either an array of JSONs or a simply an array of integers
  onChange : '&', // callback for when a user checks or unchecks a permission.
}
```